CMakeLists.txt

```diff
@@ -62,6 +62,15 @@ endif()
 # ----- HPX
 find_package(HPX REQUIRED)
 
+# In HPX 1.3.0 there is compiler matching check, but it does not check matching of build types.
+# Since it is source of headaches, waiting for HPX to add this check, we check on our own
+if (NOT "${HPX_BUILD_TYPE}" STREQUAL "${CMAKE_BUILD_TYPE}")
```

Comment:
Perhaps this can be improved to not be triggered for multi-config generators:

`if (NOT "${HPX_BUILD_TYPE}" STREQUAL "${CMAKE_BUILD_TYPE}" AND NOT CMAKE_CONFIGURATION_TYPES)`

Reply:
What about [GENERATOR_IS_MULTI_CONFIG](https://cmake.org/cmake/help/v3.16/prop_gbl/GENERATOR_IS_MULTI_CONFIG.html)?